src/ocm/components/AddHosts/day2Wizard/Day2StaticIP.tsx

```diff
@@ -0,0 +1,134 @@
+  const [confirmOnChangeView, setConfirmOnChangeView] = React.useState<boolean>(false);
+  const [viewChanged, setViewChanged] = React.useState<boolean>(false);
+  const [initialStaticIpInfo, setInitialStaticIpInfo] = React.useState<StaticIpInfo>();
+  const [formState, setFormState] = React.useState<StaticIpFormState>();
+  const { data, close } = day2DiscoveryImageDialog;
+  const cluster = data.cluster;
+  const { updateInfraEnv } = useInfraEnv(cluster.id, CpuArchitecture.x86);
+
+  React.useEffect(() => {
+    const doItAsync = async () => {
+      const { data: infraEnv } = await InfraEnvsService.getInfraEnv(
+        cluster.id,
+        CpuArchitecture.x86,
+      );
+      setInfraEnv(infraEnv);
+      infraEnv && setInitialStaticIpInfo(getStaticIpInfo(infraEnv));
+    };
+    void doItAsync();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const onChangeView = React.useCallback((view: StaticIpView) => {
+    wizardContext.onUpdateStaticIpView(view);
+    setViewChanged(true);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  if (!initialStaticIpInfo || !infraEnv) {
+    return null;
+  }
+
+  const onFormStateChange = (formState: StaticIpFormState) => {
+    const hasFilledData =
+      wizardContext.currentStepId === 'static-ip-host-configurations' || !formState.isEmpty;
+    setConfirmOnChangeView(hasFilledData);
+    setFormState(formState);
+  };
+
+  const viewProps: StaticIpViewProps = {
+    onFormStateChange,
+    infraEnv,
+    updateInfraEnv: async (params: InfraEnvUpdateParams) => {
+      const data = await updateInfraEnv(params);
+      setInfraEnv(data);
+      return data;
+    },
+    showEmptyValues: viewChanged,
+  };
+
+  const getContent = () => {
+    if (initialStaticIpInfo.view === StaticIpView.YAML && !viewChanged) {
+      return <YamlView {...viewProps} />;
+    }
+    switch (wizardContext.currentStepId) {
+      case 'static-ip-yaml-view':
+        return <YamlView {...viewProps} />;
+      case 'static-ip-host-configurations':
+        return <FormViewHosts {...viewProps} />;
+      case 'static-ip-network-wide-configurations':
+        return <FormViewNetworkWide {...viewProps} />;
+      default:
+        throw `Unexpected wizard step id ${wizardContext.currentStepId} when entering static ip page`;
+    }
+  };
+
+  const content = getContent();
+  if (!content) {
+    return null;
+  }
+
+  return (
+    <ClusterWizardStep
+      navigation={<Day2WizardNav />}
+      footer={
+        <WizardFooter
+          // eslint-disable-next-line @typescript-eslint/no-misused-promises
+          onNext={() => wizardContext.moveNext()}
+          onBack={() => wizardContext.moveBack()}
+          isNextDisabled={!formState?.isValid || formState.isAutoSaveRunning}
+          onCancel={close}
+          isSubmitting={formState?.isSubmitting}
+        />
+      }
+    >
+      {
+        <Grid hasGutter>
```

Comment:
Couldn't we reuse  `StaticIpPage` component?

Reply:
Possibly? But if would require making non-trivial changes in the `StaticIP` component - each uses different wizard context. Duplicating the code felt easier in the moment and it didn't require testing the day1 static ip. :sweat_smile: 

If you think we shouldn't have duplicate code, I can look into it in the future. But foor now, I would keep it as it is - working.

Reply:
Agree, let's try to do it in the future